### ISSUE TITLE
Distinguish between type template parameter and non-type template parameter highlightings

### ISSUE THREAD

**Issue by macrostring** [clangd/clangd#787]
Currently, type template parameters and non-type template parameters get the same highlighting kind `TemplateParameter`.

<img width="979" alt="Screen Shot" src="https://user-images.githubusercontent.com/35601156/119131741-da017c80-ba07-11eb-96c5-8f1436eac95e.png">As shown above, `_Ip` and `_Tp` are highlighted in the same color.

IMHO, it would seem to be better if `_Ip` get a highlighting of type `Variable`.

**Comment by sam-mccall** [comment]
Agree. In fact LSP calls this kind `typeParameter` (TemplateParameter is a clangd-internal name), and in practice it seems likely editors will style it as generic "type" which isn't appropriate for non-type-template params.

Any interest in sending a patch? :-)

**Comment by HighCommander4** [comment]
> IMHO, it would seem to be better if `_Ip` get a highlighting of type `Variable`.

I think it's useful to distinguish non-type template parameters from other variables, just as it's useful to distinguish type template parameters from other types.

So, I would prefer giving them a dedicated highlighting kind over mixing them into `Variable`.

**Comment by macrostring** [comment]
> So, I would prefer giving them a dedicated highlighting kind over mixing them into `Variable`.

Exactly. What I mean is that the highlighting behaviour of non-type template parameters may be like variables. As what Sam said, it's determined by the plugin/editor. My description was ambiguous. Sorry for that.
